flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorTest.java

```diff
@@ -2317,18 +2318,21 @@ private void testMaxConcurrentAttempts(int maxConcurrentAttempts) {
                     new AcknowledgeCheckpoint(graph.getJobID(), attemptID1, 1L),
                     TASK_MANAGER_LOCATION_INFO);
 
-            final Collection<ScheduledFuture<?>> periodicScheduledTasks =
-                    manuallyTriggeredScheduledExecutor.getActivePeriodicScheduledTask();
-            assertThat(periodicScheduledTasks.size()).isOne();
+            final Collection<ScheduledFuture<?>> nonPeriodicScheduledTasks =
+                    manuallyTriggeredScheduledExecutor.getActiveNonPeriodicScheduledRunnable(
+                            CheckpointCoordinator.ScheduledTrigger.class);
```

Comment:
Passing a class as the parameter to test util method appears very hacky. Is there a way to avoid doing this?

For example, can we just verify that the expected instance of runnable is found in the list of the scheduled tasks?